lib/CGameState.cpp

```diff
@@ -2187,17 +2194,21 @@ void CGameState::updateRumor()
 	while(!rumor.update(rumorId, rumorExtra));
 }
 
-bool CGameState::isVisible(int3 pos, PlayerColor player)
+bool CGameState::isVisible(int3 pos, boost::optional<PlayerColor> player) const
```

Comment:
If callback is owner by server - server will use this method for "is object visible for player X"
If callback is owner by client - it will use this method for "is object visible for me" queries BUT it will not provide player ID - it will use player ID stored in CCallback class

This change is about making methods with same name virtual - to avoid having classes with methods that use same name but different signature than virtual method in base class

Reply:
Instead I would suggest to use default argument with PlayerColor::CANNOT_DETERMINE or other constant

Reply:
I'll probably leave it as it.
First of all, this boost::optional was present before;
https://github.com/vcmi/vcmi/blob/develop/lib/CGameInfoCallback.h#L138
Second, boost::optional is frequently used for "self-checks" - CGameInfoCallback has its own boost::optional<PlayerColor> which is set when client creates PlayerInterface. Don't want massive changes which could break who knows what.

I agree that this is... weird design decision, but it is outside of "fix compiler warnings" scope